cpp/CMakeLists.txt

```diff
@@ -251,7 +251,7 @@ endif()
 
 if(RAFT_NVTX)
   # This enables NVTX within the project with no option to disable it downstream.
-  target_link_libraries(raft INTERFACE CUDA::nvToolsExt)
+  target_link_libraries(raft INTERFACE CUDA::nvtx3)
```

Comment:
This has been added to resolve the following deprecation warning:
```
The library that is being linked to, CUDA::nvToolsExt, is marked as being
deprecated by the owner.  The message provided by the developer is:

nvToolsExt has been superseded by nvtx3 since CUDA 10.0 and CMake 3.25.
Use CUDA::nvtx3 and include <nvtx3/nvToolsExt.h> instead.
```